agent/agent_worker.go

```diff
@@ -412,7 +412,9 @@ func (a *AgentWorker) AcquireAndRunJob(jobId string) error {
 
 	// If `acquiredJob` is nil, then the job was never acquired
 	if acquiredJob == nil {
-		return fmt.Errorf("Failed to acquire job: %v", err)
+		a.logger.Warn("Failed to acquire job: %v", err)
+		// Return nil so that the AgentPool disconnnects this one-shot agent
+		return nil
```

Comment:
Oh this is a good find! I need to double check the code, but should we be returning `err` here perhaps?

Reply:
I think returning the `err` here will cause the same behaviour as before.

Where we call `s.Break()` above :point_up:, the case we want to disconnect the agent under, we know `err` is non-nil. That function returns the non-nil `err` to `retry.Do` which then returns it to us. If we return it from here the `AgentPool` won't disconnect us and leaves the agent hanging until it is pruned.

Maybe this is actually a problem with the `AgentPool` and we want it to try disconnecting the agent when `Agent.Start` returns error or no?

Reply:
> Maybe this is actually a problem with the AgentPool and we want it to try disconnecting the agent when Agent.Start returns error or no?

I reckon that might be the winner. Are you ok to make that change? (I appreciate you bug finding and fixing, so I don't want to push my luck with code change requests :P)